File: updateReadMe.js
	const fs = require('fs');
	const path = require('path');

	const fileName = path.basename(__filename);

	const ignoreList = [
	fileName,
	".git",
	"_template",
	"README.md"
	];

	const reg = new RegExp(ignoreList.join("|"), "i");

	let filesToLog = "";

	let fileContent = "";

	fileContent += "# Index \n"; 

	files = fs.readdirSync("./", {withFileTypes:true});
	for(f of files)
	{
		if(!f.isDirectory())
			continue;
		
		if(!reg.test(f.name))
		{
			filesToLog += `[${f.name}](https://github.com/noenarcisse/MultiLangStudy/tree/main/${f.name.replaceAll(" ", "%20")})  <br> \n`;
		}
	}

	fileContent += filesToLog;

	console.log(fileContent);


	fs.writeFileSync("./README.md", fileContent);

File: updatereadme.nim
import os
import strformat
import uri

proc formatMarkdown(dirname : string) : string =
    &"[{dirname}](https://github.com/noenarcisse/MultiLangStudy/tree/main/{encodeUrl dirname})<br>\n"

var f : File = nil
let ignored = @[".git", ".vscode", "_template"]
let readme_file = "README.md"
let header = """
# Index
Liste des catégories : <br>
"""
var content : string

for e in walkDir(".") :
    if e.kind != PathComponent.pcDir : continue
    var dirname = extractFilename(e.path)
    if dirname in ignored : continue
    content&=formatMarkdown dirname

if open(f, readme_file, FileMode.fmWrite) :
    defer : close f
    f&=header&"\n"&content #write non trycatch
else :
    echo "Outch, opening file failed"